### ISSUE TITLE
Error with signed/unsigned natives when printing a buf64 that contains 2**64-1

### ISSUE THREAD

**Issue by MasterDuke17** [rakudo/rakudo#5348]
`raku -e 'my $bufb = buf64.new(3, 1, 18446744073709551615);'` survives, but `raku --ll-exception -e 'my $bufb = buf64.new(3, 1, 18446744073709551615); say $bufb'` dies with:
```
Cannot unbox 64 bit wide bigint into native integer
   at SETTING::src/core.c/Buf.pm6:688  (/home/dan/Source/perl6/install/share/perl6/runtime/CORE.c.setting.moarvm:gist)
 from SETTING::src/core.c/io_operators.pm6:91  (/home/dan/Source/perl6/install/share/perl6/runtime/CORE.c.setting.moarvm:say)
 from -e:1  (<ephemeral file>:<unit>)
 from -e:1  (<ephemeral file>:<unit-outer>)
 from NQP::src/HLL/Compiler.nqp:196  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:eval)
 from NQP::src/HLL/Compiler.nqp:309  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:)
 from NQP::src/HLL/Compiler.nqp:364  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:command_eval)
 from src/Perl6/Compiler.nqp:229  (/home/dan/Source/perl6/install/share/perl6/lib/Perl6/Compiler.moarvm:command_eval)
 from NQP::src/HLL/Compiler.nqp:289  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:command_line)
 from src/main.nqp:81  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:MAIN)
 from src/main.nqp:63  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<mainline>)
 from <unknown>:1  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<main>)
 from <unknown>:1  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<entry>)
```
```
[dan@alexandria raku-bench]$ gdb --args raku -e 'my $bufb = buf64.new(3, 1, 18446744073709551615); say $bufb'
GNU gdb (GDB) 13.2
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from raku...
(gdb) b MVM_exception_throw_adhoc
Function "MVM_exception_throw_adhoc" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (MVM_exception_throw_adhoc) pending.
(gdb) r
Starting program: /home/dan/Source/perl6/install/bin/raku -e my\ \$bufb\ =\ buf64.new\(3,\ 1,\ 18446744073709551615\)\;\ say\ \$bufb
Downloading separate debug info for /lib64/ld-linux-x86-64.so.2
Downloading separate debug info for system-supplied DSO at 0x7ffff7fc8000                                                                                                                                                                         
Downloading separate debug info for /usr/lib/libc.so.6                                                                                                                                                                                            
[Thread debugging using libthread_db enabled]                                                                                                                                                                                                     
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Downloading separate debug info for /usr/lib/libm.so.6
Downloading separate debug info for /usr/lib/libzstd.so.1                                                                                                                                                                                         
[New Thread 0x7ffff712c6c0 (LWP 152263)]                                                                                                                                                                                                          

Thread 1 "raku" hit Breakpoint 1, MVM_exception_throw_adhoc (tc=tc@entry=0x35308020180, messageFormat=messageFormat@entry=0x7ffff79c79d8 "Cannot unbox %d bit wide bigint into native integer") at src/core/exceptions.c:882
882     MVM_NO_RETURN void MVM_exception_throw_adhoc(MVMThreadContext *tc, const char *messageFormat, ...) {
(gdb) bt
#0  MVM_exception_throw_adhoc (tc=tc@entry=0x35308020180, messageFormat=messageFormat@entry=0x7ffff79c79d8 "Cannot unbox %d bit wide bigint into native integer") at src/core/exceptions.c:882
#1  0x00007ffff789a2eb in mp_get_int64 (a=0x3530bff6d40, tc=0x35308020180) at src/6model/reprs/P6bigint.c:23
#2  get_int (st=<optimized out>, root=<optimized out>, data=<optimized out>, tc=0x35308020180) at src/6model/reprs/P6bigint.c:117
#3  get_int (tc=0x35308020180, st=<optimized out>, root=<optimized out>, data=<optimized out>) at src/6model/reprs/P6bigint.c:113
#4  0x00007ffff7885915 in MVM_6model_container_decont_i (tc=<optimized out>, cont=<optimized out>, res=0x353084825d0) at src/6model/containers.c:895
#5  0x00007ffff782d208 in MVM_interp_run (tc=0x35308020180, initial_invoke=0x7ffff79c79d8, initial_invoke@entry=0x7ffff795d700 <toplevel_initial_invoke>, invoke_data=0x7ffff79c79d8, 
    invoke_data@entry=0x7ffff795d700 <toplevel_initial_invoke>, outer_runloop=0x555555562ca0, outer_runloop@entry=0x0) at src/core/interp.c:4282
#6  0x00007ffff795e7b5 in MVM_vm_run_file (instance=instance@entry=0x35308010000, filename=filename@entry=0x555555559520 "/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm") at src/moar.c:505
#7  0x00005555555557c8 in main (argc=<optimized out>, argv=<optimized out>) at src/vm/moar/runner/main.c:474
(gdb) call MVM_dump_backtrace(tc)
   at SETTING::src/core.c/Buf.pm6:688  (/home/dan/Source/perl6/install/share/perl6/runtime/CORE.c.setting.moarvm:gist)
 from SETTING::src/core.c/io_operators.pm6:91  (/home/dan/Source/perl6/install/share/perl6/runtime/CORE.c.setting.moarvm:say)
 from -e:1  (<ephemeral file>:<unit>)
 from -e:1  (<ephemeral file>:<unit-outer>)
 from NQP::src/HLL/Compiler.nqp:196  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:eval)
 from NQP::src/HLL/Compiler.nqp:309  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:)
 from NQP::src/HLL/Compiler.nqp:306  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:command_eval)
 from src/Perl6/Compiler.nqp:229  (/home/dan/Source/perl6/install/share/perl6/lib/Perl6/Compiler.moarvm:command_eval)
 from NQP::src/HLL/Compiler.nqp:289  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:command_line)
 from src/main.nqp:81  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:MAIN)
 from src/main.nqp:63  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<mainline>)
 from <unknown>:1  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<main>)
 from <unknown>:1  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<entry>)
(gdb) call MVM_dump_bytecode(tc)
00000      checkarity         1, 1
00001      param_rp_o         loc_6_obj, 0
00002      dispatch_o         loc_9_obj, 'lang-hllize', Callsite_1, loc_6_obj
00003      set                loc_6_obj, loc_9_obj
00004      decont             loc_9_obj, loc_6_obj
00005      set                loc_32_obj, loc_9_obj
00006      wval               loc_9_obj, 1, 20747
00007      istype             loc_27_int, loc_32_obj, loc_9_obj
00008      assertparamcheck   loc_27_int
00009      isconcrete         loc_27_int, loc_32_obj
00010      assertparamcheck   loc_27_int
00011      set                loc_0_obj, loc_32_obj
00012      param_sn           loc_7_obj
     annotation: SETTING::src/core.c/Buf.pm6:681
00013      wval               loc_8_obj, 0, 28
00014      create             loc_8_obj, loc_8_obj
00015      wval               loc_9_obj, 0, 28
00016      wval               loc_10_obj, 1, 118
00017      bindattr_o         loc_8_obj, loc_9_obj, '$!descriptor', loc_10_obj, -1
00018      wval               loc_10_obj, 0, 28
00019      wval               loc_9_obj, 0, 24
00020      bindattr_o         loc_8_obj, loc_10_obj, '$!value', loc_9_obj, -1
00021      set                loc_4_obj, loc_8_obj
     annotation: SETTING::src/core.c/Buf.pm6:683
00022      getlex             loc_10_obj, lex_Frame_13252_lex2_obj
00023      set                loc_9_obj, loc_10_obj
00024      decont             loc_10_obj, loc_9_obj
00025      gethow             loc_10_obj, loc_10_obj
00026      decont             loc_11_obj, loc_10_obj
00027      const_s            loc_12_str, 'nativesize'
00028      dispatch_o         loc_10_obj, 'lang-meth-call', Callsite_2, loc_11_obj, loc_12_str, loc_10_obj, loc_9_obj
00029      dispatch_o         loc_11_obj, 'lang-hllize', Callsite_1, loc_10_obj
00030      set                loc_8_obj, loc_11_obj
00031      decont             loc_11_obj, loc_8_obj
00032      const_s            loc_12_str, 'defined'
00033      dispatch_o         loc_11_obj, 'lang-meth-call', Callsite_6, loc_11_obj, loc_12_str, loc_8_obj
00034      decont             loc_14_obj, loc_11_obj
00035      dispatch_i         loc_15_int, 'boot-boolify', Callsite_1, loc_14_obj
00036      unless_i           loc_15_int, label_1(00039)
00037      set                loc_13_obj, loc_8_obj
00038      goto               label_2(00041)
     label_1:
00039      getlex             loc_10_obj, lex_Frame_4126_lex206_obj
00040      set                loc_13_obj, loc_10_obj
     label_2:
00041      decont_i           loc_15_int, loc_13_obj
00042      const_i64_16       loc_16_int, 4
00043      div_i              loc_16_int, loc_15_int, loc_16_int
00044      hllboxtype_i       loc_10_obj
00045      box_i              loc_10_obj, loc_16_int, loc_10_obj
00046      set                loc_11_obj, loc_10_obj
00047      if_i               loc_16_int, label_3(00050)
00048      wval               loc_13_obj, 1, 117
00049      set                loc_11_obj, loc_13_obj
     label_3:
00050      decont_i           loc_16_int, loc_11_obj
00051      bindlex            lex_Frame_13309_lex10_int, loc_16_int
     annotation: SETTING::src/core.c/Buf.pm6:685
00052      elems              loc_17_int, loc_0_obj
00053      set                loc_15_int, loc_17_int
00054      const_i64_16       loc_18_int, 200
00055      getlex             loc_19_int, lex_Frame_13309_lex10_int
00056      div_i              loc_19_int, loc_18_int, loc_19_int
00057      set                loc_17_int, loc_19_int
00058      cmp_i              loc_19_int, loc_15_int, loc_17_int
00059      const_i64_16       loc_18_int, 0
00060      gt_i               loc_18_int, loc_19_int, loc_18_int
00061      unless_i           loc_18_int, label_4(00064)
00062      set                loc_19_int, loc_17_int
00063      goto               label_5(00065)
     label_4:
00064      set                loc_19_int, loc_15_int
     label_5:
00065      hllboxtype_i       loc_8_obj
00066      box_i              loc_8_obj, loc_19_int, loc_8_obj
00067      dispatch_o         loc_9_obj, 'raku-rv-decont', Callsite_1, loc_8_obj
00068      decont_i           loc_19_int, loc_9_obj
00069      bindlex            lex_Frame_13309_lex9_int, loc_19_int
     annotation: SETTING::src/core.c/Buf.pm6:686
00070      const_i64_16       loc_15_int, -1
00071      bindlex            lex_Frame_13309_lex8_int, loc_15_int
     annotation: SETTING::src/core.c/Buf.pm6:687
00072      bootstrarray       loc_9_obj
00073      create             loc_9_obj, loc_9_obj
00074      set                loc_3_obj, loc_9_obj
     annotation: SETTING::src/core.c/Buf.pm6:688
00075      getlex             loc_8_obj, lex_Frame_13252_lex2_obj
00076      set                loc_9_obj, loc_8_obj
00077      decont             loc_8_obj, loc_9_obj
00078      gethow             loc_8_obj, loc_8_obj
00079      decont             loc_11_obj, loc_8_obj
00080      const_s            loc_12_str, 'unsigned'
00081      dispatch_o         loc_8_obj, 'lang-meth-call', Callsite_2, loc_11_obj, loc_12_str, loc_8_obj, loc_9_obj
00082      dispatch_o         loc_11_obj, 'lang-hllize', Callsite_1, loc_8_obj
00083   -> decont_i           loc_17_int, loc_11_obj
00084      set                loc_2_int, loc_17_int
     label_6:
00085      getlex             loc_17_int, lex_Frame_13309_lex8_int
00086      const_i64_16       loc_18_int, 1
00087      add_i              loc_18_int, loc_17_int, loc_18_int
00088      bindlex            lex_Frame_13309_lex8_int, loc_18_int
00089      getlex             loc_17_int, lex_Frame_13309_lex9_int
00090      lt_i               loc_17_int, loc_18_int, loc_17_int
00091      unless_i           loc_17_int, label_11(00138)
00092      osrpoint           
00093      const_i64_16       loc_18_int, 0
00094      ne_i               loc_18_int, loc_2_int, loc_18_int
00095      unless_i           loc_18_int, label_7(00102)
00096      getlex             loc_20_int, lex_Frame_13309_lex8_int
00097      atpos_u            loc_21_uint, loc_0_obj, loc_20_int
00098      hllboxtype_i       loc_11_obj
00099      box_u              loc_11_obj, loc_21_uint, loc_11_obj
00100      set                loc_9_obj, loc_11_obj
00101      goto               label_8(00107)
     label_7:
00102      getlex             loc_20_int, lex_Frame_13309_lex8_int
00103      atpos_i            loc_20_int, loc_0_obj, loc_20_int
00104      hllboxtype_i       loc_8_obj
00105      box_i              loc_8_obj, loc_20_int, loc_8_obj
00106      set                loc_9_obj, loc_8_obj
     label_8:
00107      decont_i           loc_18_int, loc_9_obj
00108      bindlex            lex_Frame_13309_lex7_int, loc_18_int
00109      bootstrarray       loc_9_obj
00110      create             loc_9_obj, loc_9_obj
00111      set                loc_1_obj, loc_9_obj
00112      getlex             loc_20_int, lex_Frame_13309_lex10_int
00113      bindlex            lex_Frame_13309_lex6_int, loc_20_int
     label_9:
00114      getlex             loc_22_int, lex_Frame_13309_lex6_int
00115      const_i64_16       loc_23_int, 0
00116      gt_i               loc_23_int, loc_22_int, loc_23_int
00117      unless_i           loc_23_int, label_10(00134)
00118      osrpoint           
00119      getlex             loc_9_obj, lex_Frame_13252_lex0_obj
00120      getlex             loc_22_int, lex_Frame_13309_lex7_int
00121      const_i64_16       loc_24_int, 15
00122      band_i             loc_24_int, loc_22_int, loc_24_int
00123      atpos_s            loc_12_str, loc_9_obj, loc_24_int
00124      unshift_s          loc_1_obj, loc_12_str
00125      getlex             loc_24_int, lex_Frame_13309_lex7_int
00126      const_i64_16       loc_22_int, 4
00127      brshift_i          loc_22_int, loc_24_int, loc_22_int
00128      bindlex            lex_Frame_13309_lex7_int, loc_22_int
00129      getlex             loc_24_int, lex_Frame_13309_lex6_int
00130      const_i64_16       loc_25_int, 1
00131      sub_i              loc_25_int, loc_24_int, loc_25_int
00132      bindlex            lex_Frame_13309_lex6_int, loc_25_int
00133      goto               label_9(00114)
     label_10:
00134      const_s            loc_26_str, ''
00135      join               loc_26_str, loc_26_str, loc_1_obj
00136      push_s             loc_3_obj, loc_26_str
00137      goto               label_6(00085)
     annotation: SETTING::src/core.c/Buf.pm6:711
     label_11:
00138      elems              loc_24_int, loc_0_obj
00139      getlex             loc_27_int, lex_Frame_13309_lex9_int
00140      gt_i               loc_27_int, loc_24_int, loc_27_int
00141      unless_i           loc_27_int, label_12(00148)
00142      const_s            loc_28_str, '...'
00143      push_s             loc_3_obj, loc_28_str
00144      hllboxtype_s       loc_14_obj
00145      box_s              loc_14_obj, loc_28_str, loc_14_obj
00146      set                loc_13_obj, loc_14_obj
00147      goto               label_13(00150)
     label_12:
00148      wval               loc_9_obj, 1, 106
00149      set                loc_13_obj, loc_9_obj
     label_13:
00150      dispatch_v         'raku-sink', Callsite_1, loc_13_obj
     annotation: SETTING::src/core.c/Buf.pm6:714
00151      const_s            loc_28_str, ''
00152      bootstrarray       loc_13_obj
00153      create             loc_13_obj, loc_13_obj
00154      const_i64_16       loc_27_int, 4
00155      setelemspos        loc_13_obj, loc_27_int
00156      const_i64_16       loc_27_int, 0
00157      setelemspos        loc_13_obj, loc_27_int
00158      set                loc_9_obj, loc_0_obj
00159      decont             loc_29_obj, loc_9_obj
00160      gethow             loc_29_obj, loc_29_obj
00161      decont             loc_30_obj, loc_29_obj
00162      const_s            loc_31_str, 'name'
00163      dispatch_o         loc_29_obj, 'lang-meth-call', Callsite_2, loc_30_obj, loc_31_str, loc_29_obj, loc_9_obj
00164      dispatch_o         loc_30_obj, 'lang-hllize', Callsite_1, loc_29_obj
00165      decont_s           loc_31_str, loc_30_obj
00166      push_s             loc_13_obj, loc_31_str
00167      const_s            loc_31_str, ':0x<'
00168      push_s             loc_13_obj, loc_31_str
00169      const_s            loc_31_str, ' '
00170      join               loc_31_str, loc_31_str, loc_3_obj
00171      push_s             loc_13_obj, loc_31_str
00172      const_s            loc_31_str, '>'
00173      push_s             loc_13_obj, loc_31_str
00174      join               loc_28_str, loc_28_str, loc_13_obj
00175      hllboxtype_s       loc_9_obj
00176      box_s              loc_9_obj, loc_28_str, loc_9_obj
00177      dispatch_o         loc_13_obj, 'raku-rv-decont', Callsite_1, loc_9_obj
00178      return_o           loc_13_obj
(gdb)
```

**Comment by MasterDuke17** [comment]
Similar error for `raku -e 'my $bufb = buf64.new(3, 1, 18446744073709551614); $bufb[2] = $bufb[2] + 1;'`:
```
[dan@alexandria raku-bench]$ gdb --args raku -e 'my $bufb = buf64.new(3, 1, 18446744073709551615); $bufb[2] = $bufb[2] + 1;'
GNU gdb (GDB) 13.2
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from raku...
(gdb) b MVM_exception_throw_adhoc
Function "MVM_exception_throw_adhoc" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (MVM_exception_throw_adhoc) pending.
(gdb) r
Starting program: /home/dan/Source/perl6/install/bin/raku -e my\ \$bufb\ =\ buf64.new\(3,\ 1,\ 18446744073709551615\)\;\ \$bufb\[2\]\ =\ \$bufb\[2\]\ +\ 1\;
[Thread debugging using libthread_db enabled]                                                                                                                                                                                                     
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7ffff712c6c0 (LWP 160813)]                                                                                                                                                                                                          

Thread 1 "raku" hit Breakpoint 1, MVM_exception_throw_adhoc (tc=tc@entry=0x4b6e8020180, messageFormat=messageFormat@entry=0x7ffff79bf8a8 "Cannot unbox %d bit wide bigint into native integer") at src/core/exceptions.c:882
882     MVM_NO_RETURN void MVM_exception_throw_adhoc(MVMThreadContext *tc, const char *messageFormat, ...) {
(gdb) call MVM_dump_backtrace(tc)
   at SETTING::src/core.c/Buf.pm6:1053  (/home/dan/Source/perl6/install/share/perl6/runtime/CORE.c.setting.moarvm:ASSIGN-POS)
 from SETTING::src/core.c/array_slice.pm6:17  (/home/dan/Source/perl6/install/share/perl6/runtime/CORE.c.setting.moarvm:postcircumfix:<[ ]>)
 from -e:1  (<ephemeral file>:<unit>)
 from -e:1  (<ephemeral file>:<unit-outer>)
 from NQP::src/HLL/Compiler.nqp:196  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:eval)
 from NQP::src/HLL/Compiler.nqp:309  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:)
 from NQP::src/HLL/Compiler.nqp:306  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:command_eval)
 from src/Perl6/Compiler.nqp:229  (/home/dan/Source/perl6/install/share/perl6/lib/Perl6/Compiler.moarvm:command_eval)
 from NQP::src/HLL/Compiler.nqp:289  (/home/dan/Source/perl6/install/share/nqp/lib/NQPHLL.moarvm:command_line)
 from src/main.nqp:81  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:MAIN)
 from src/main.nqp:63  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<mainline>)
 from <unknown>:1  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<main>)
 from <unknown>:1  (/home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm:<entry>)
(gdb) call MVM_dump_bytecode(tc)
00000      checkarity         3, 3
00001      param_rp_o         loc_5_obj, 0
00002      dispatch_o         loc_10_obj, 'lang-hllize', Callsite_1, loc_5_obj
00003      set                loc_5_obj, loc_10_obj
00004      decont             loc_10_obj, loc_5_obj
00005      set                loc_16_obj, loc_10_obj
00006      wval               loc_10_obj, 1, 21789
00007      istype             loc_13_int, loc_16_obj, loc_10_obj
00008      assertparamcheck   loc_13_int
00009      isconcrete         loc_13_int, loc_16_obj
00010      assertparamcheck   loc_13_int
00011      set                loc_0_obj, loc_16_obj
00012      param_rp_o         loc_6_obj, 1
00013      dispatch_o         loc_10_obj, 'lang-hllize', Callsite_1, loc_6_obj
00014      set                loc_6_obj, loc_10_obj
00015      decont             loc_10_obj, loc_6_obj
00016      set                loc_17_obj, loc_10_obj
00017      wval               loc_10_obj, 0, 42
00018      istype             loc_13_int, loc_17_obj, loc_10_obj
00019      assertparamcheck   loc_13_int
00020      isconcrete         loc_13_int, loc_17_obj
00021      assertparamcheck   loc_13_int
00022      set                loc_3_obj, loc_17_obj
00023      param_rp_o         loc_7_obj, 2
00024      dispatch_o         loc_10_obj, 'lang-hllize', Callsite_1, loc_7_obj
00025   -> set                loc_7_obj, loc_10_obj
00026      set                loc_4_obj, loc_7_obj
00027      param_sn           loc_8_obj
     annotation: SETTING::src/core.c/Buf.pm6:1052
00028      wval               loc_9_obj, 0, 28
00029      create             loc_9_obj, loc_9_obj
00030      wval               loc_10_obj, 0, 28
00031      wval               loc_11_obj, 1, 118
00032      bindattr_o         loc_9_obj, loc_10_obj, '$!descriptor', loc_11_obj, -1
00033      wval               loc_11_obj, 0, 28
00034      wval               loc_10_obj, 0, 24
00035      bindattr_o         loc_9_obj, loc_11_obj, '$!value', loc_10_obj, -1
00036      set                loc_1_obj, loc_9_obj
     annotation: SETTING::src/core.c/Buf.pm6:1053
00037      decont_i           loc_12_int, loc_3_obj
00038      const_i64_16       loc_13_int, 0
00039      lt_i               loc_13_int, loc_12_int, loc_13_int
00040      unless_i           loc_13_int, label_1(00048)
00041      const_s            loc_14_str, '::?CLASS'
00042      getlexperinvtype_o loc_9_obj, loc_14_str
00043      const_s            loc_14_str, 'fail-range'
00044      dispatch_o         loc_10_obj, 'raku-meth-private', Callsite_2, loc_9_obj, loc_14_str, loc_0_obj, loc_3_obj
00045      dispatch_o         loc_9_obj, 'lang-hllize', Callsite_1, loc_10_obj
00046      set                loc_10_obj, loc_9_obj
00047      goto               label_2(00054)
     label_1:
00048      decont_i           loc_12_int, loc_3_obj
00049      decont_u           loc_15_uint, loc_4_obj
00050      bindpos_u          loc_0_obj, loc_12_int, loc_15_uint
00051      hllboxtype_i       loc_9_obj
00052      box_u              loc_9_obj, loc_15_uint, loc_9_obj
00053      set                loc_10_obj, loc_9_obj
     label_2:
00054      dispatch_o         loc_11_obj, 'raku-rv-decont', Callsite_1, loc_10_obj
00055      return_o           loc_11_obj
(gdb)
```

**Comment by MasterDuke17** [comment]
This error found by desmond/demondthesecond over in #raku-beginner.

**Comment by lizmat** [comment]
I cannot reproduce `'my $bufb = buf64.new(3, 1, 18446744073709551614); $bufb[2] = $bufb[2] + 1;`  Works fine for me.

**Comment by lizmat** [comment]
Added test with https://github.com/Raku/roast/commit/0529a785e9

Closing this issue again as I can't reproduce the second case.  Please open a new one if there is a reproducible problem!  Thanks!

**Comment by MasterDuke17** [comment]
> I cannot reproduce `'my $bufb = buf64.new(3, 1, 18446744073709551614); $bufb[2] = $bufb[2] + 1;` Works fine for me.

Oops, that was supposed to be `my $bufb = buf64.new(3, 1, 18446744073709551615); $bufb[2] = $bufb[2] + 1;` (note the large value ends in 5, not 4). I accidentally copied the wrong line from my history.

**Comment by lizmat** [comment]
I think that is another issue, and one that may warrant a problem-solving issue.
